# One-Sample Z-Test for Proportions: pandas-dev/pandas Statistical Audit

### Research Questions:
* **RQ1:** Apakah proporsi keberhasilan merge Pull Request (PR) secara signifikan berbeda dari standar klaim historis sebesar 60%? **
* **RQ2:** Berapa rentang estimasi probabilitas PR berhasil di-merge pada tingkat kepercayaan 95%? **
* **RQ3:** Apakah rata-rata issue mingguan berubah secara signifikan setelah adanya major release? *

**Member:** [Nama Kamu] — Inference Analyst (Member D)

---

### AI Usage Disclosure
**Anggota:** [Nama Kamu] – Inference Analyst | **Alat yang digunakan:** Gemini

| Task / Tool | Prompt Summary | Output Modified |
| :--- | :--- | :--- |
| **Scaffolding Prosedur Z-Test** | Gemini | Meminta boilerplate fungsi z-test satu sampel menggunakan scipy.stats untuk pengujian proporsi biner. | **Ya.** Struktur kode dipisah ke dalam modul `src/hypothesis.py`, mengisolasi logika fungsi komputasi dari visualisasi notebook. |
| **Penyusunan Narasi 6-Step** | Gemini | Meminta kerangka penulisan formal 6-Step Hypothesis Testing berdasarkan rujukan literatur statistika dengan 3 RQ. | **Ya.** Narasi dirombak total secara mandiri untuk menjawab RQ1 secara spesifik menggunakan parameter n dan x_bar dari Member B. |
| **Troubleshooting Kernel Jupyter** | Gemini | Meminta bantuan penanganan error hilangnya environment interpreter Python dan paket ipykernel di VS Code. | **Ya.** Instalasi dan pemilihan versi Python dilakukan secara manual pada local machine hingga visualisasi berhasil dirender. |

> **Pernyataan Mandiri:** Seluruh keputusan interpretasi statistik, pengujian asumsi Central Limit Theorem (CLT), dan pengambilan keputusan menolak H0 dirumuskan, ditulis, dan diverifikasi kembali secara mandiri tanpa modifikasi mentah dari kecerdasan buatan.

---

## Analisis Inferensia: Pengujian Hipotesis (Z-Test)

### 1. Persiapan Data & Eksekusi Komputasi
Mengambil hasil observasi dari Estimation Analyst (Member B) dengan total sampel $n = 4042$ dan proporsi sukses $\bar{x} = 0.62098$.

In [3]:
import sys
sys.path.append("../")
import numpy as np
from src.hypothesis import z_test_one_sample

# Data hasil panen dari Member B & C
n = 4042
x_bar = 0.62098  # Proporsi Pull Request yang berhasil di-merge (62.10%)

# Standar Deviasi untuk distribusi proporsi (Bernoulli): sqrt(p * (1 - p))
sigma = np.sqrt(x_bar * (1 - x_bar))

# Nilai hipotesis acuan (Klaim standar historis 60%)
mu0 = 0.60000 

# Eksekusi fungsi Z-Test dari file src/hypothesis.py
hasil = z_test_one_sample(
    x_bar=x_bar, 
    mu0=mu0, 
    sigma=sigma, 
    n=n, 
    alternative='two-sided', 
    alpha=0.05
)

print("=== FREQUENTIST HYPOTHESIS TESTING (Z-TEST) ===")
print(f"Total Sampel (n)     : {n}")
print(f"Proporsi Sukses (x)  : {x_bar:.5f}")
print(f"Z-Statistic Calculated: {hasil['z_stat']:.4f}")
print(f"P-Value Calculated   : {hasil['p_value']:.4f}")
print(f"Statistical Decision : {hasil['decision']}")

=== FREQUENTIST HYPOTHESIS TESTING (Z-TEST) ===
Total Sampel (n)     : 4042
Proporsi Sukses (x)  : 0.62098
Z-Statistic Calculated: 2.7494
P-Value Calculated   : 0.0060
Statistical Decision : Reject H0


### Prosedur Uji Hipotesis (6-Step Procedure) - Member D

**Step 1: State Hypotheses**
* $H_0: \mu = 0.60$ (Proporsi pull request yang berhasil di-merge sama dengan 60%)
* $H_a: \mu \neq 0.60$ (Proporsi pull request yang berhasil di-merge secara signifikan berbeda dari 60%)

**Step 2: Assumptions Check**
* Sampel data bersifat independen yang ditarik langsung dari log historis repositori `pandas-dev/pandas`.
* Ukuran sampel sangat masif ($n = 4042 \ge 30$). Berdasarkan *Central Limit Theorem* (CLT), distribusi sampling dari rata-rata sampel dapat diasumsikan mendekati distribusi normal sehingga pendekatan uji Z-Test satu sampel ini valid digunakan.

**Step 3: Test Statistic Calculation**
* Menggunakan formula uji Z untuk proporsi sesuai dengan rujukan teori Tsun (2020):
  $$Z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}$$
* Berdasarkan eksekusi modul komputasi `src/hypothesis.py`, nilai uji statistik $Z$ berhasil dikalkulasi secara presisi sebesar **2.7494**.

**Step 4: P-value and Critical Region**
* Pengujian menggunakan metode dua arah (*two-sided*) dengan tingkat signifikansi $\alpha = 0.05$. Batas wilayah kritis penolakan berdasarkan tabel distribusi normal adalah $Z < -1.96$ atau $Z > 1.96$.

**Step 5: Statistical Decision**
* Melalui komputasi berbasis paket `scipy.stats`, diperoleh *p-value* sebesar **0.0060**. Karena *p-value* ($0.0060$) < $\alpha$ ($0.05$), maka keputusan statistik yang diambil adalah **Reject $H_0$**.

**Step 6: Conclusion & Interpretation**
* Dengan tingkat signifikansi $\alpha = 0.05$, terdapat bukti statistik yang kuat untuk menyatakan bahwa proporsi pull request yang berhasil di-merge pada repositori Pandas secara signifikan berbeda dari angka asumsi 60%. Performa aktual yang menyentuh angka 62.09% merefleksikan tingkat penerimaan kontribusi kode yang sangat sehat, aktif, dan konsisten dari tim core maintainer open-source Pandas dalam mengelola proyek global.